src/ucp/core/ucp_request.inl

```diff
@@ -659,7 +659,7 @@ ucp_request_complete_am_recv(ucp_request_t *req, ucs_status_t status)
                   req->recv.length, ucs_status_string(status));
     UCS_PROFILE_REQUEST_EVENT(req, "complete_recv", status);
 
-    if (req->recv.am.desc->flags & UCP_RECV_DESC_FLAG_RNDV) {
+    if (req->recv.am.desc->flags & UCP_RECV_DESC_FLAG_INTIALIZED) {
```

Comment:
can we check UCP_RECV_DESC_FLAG_RECV_STARTED instead of UCP_RECV_DESC_FLAG_INTIALIZED?
to remove UCP_RECV_DESC_FLAG_INTIALIZED flag, or make it for assertions only

Reply:
no, because `UCP_RECV_DESC_FLAG_RECV_STARTED` indicates that `ucp_am_recv_data_nbx` is called, while I also need to distinguish the case whether it is safe to release desc (i.e. we are not on the user callback stack already)